src/Rules/UselessConstructs/NoUselessValueFunctionCallsRule.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NunoMaduro\Larastan\Rules\UselessConstructs;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+
+/**
+ * @implements Rule<FuncCall>
+ */
+class NoUselessValueFunctionCallsRule implements Rule
+{
+    public function getNodeType(): string
+    {
+        return FuncCall::class;
+    }
+
+    public function processNode(Node $node, Scope $scope): array
+    {
+        /** @var FuncCall $node */
+        if (! $node->name instanceof Node\Name) {
+            return [];
+        }
+
+        if ($node->name->toString() !== 'value') {
+            return [];
+        }
+
+        $args = $node->getArgs();
+        if ($args[0]->value instanceof Node\Expr\Closure) {
```

Comment:
We need to check if `count($args)` is greater than 1 before here.

Also using the `instanceof` check here with the `Closure` is wrong here. This wouldn't work for example union or intersection types. Instead we should ask the type system about this. You can use `$scope->getType($args[0]->value)` to get the type for the variable. And use `isSuperTypeOf` to compare the types. You can read more about the type system [here.](https://phpstan.org/developing-extensions/type-system)